app/src/test/java/com/android/bookswap/data/source/network/ImageToDataSourceTest.kt

```diff
@@ -0,0 +1,158 @@
+package com.android.bookswap.data.source.network
+
+import com.android.bookswap.data.source.api.ApiService
+import io.mockk.every
+import io.mockk.invoke
+import io.mockk.mockk
+import io.mockk.verify
+import junit.framework.Assert.assertTrue
+import junit.framework.TestCase.assertEquals
+import junit.framework.TestCase.fail
+import net.bytebuddy.implementation.InvokeDynamic.lambda
+import org.junit.Before
+import org.junit.Test
+
+class ImageToDataSourceTest {
+  private lateinit var apiService: ApiService
+  private lateinit var imageToDataSource: ImageToDataSource
+  private val imageUrl = "https://example.com/book-cover.jpg"
+  private val expectedPrompt =
+      """
+        You are an AI designed to analyze images of book covers. Given an image of a book's back cover in URL format, extract the following information in a structured format (JSON) with the specified fields: title, author, description, language, and ISBN. Please ensure that:
```

Comment:
This prompt is now written twice, and is quite long, you might want to store it in an xml as a constant string or something like that, it will make it easier to modify as well, and it is quite a lot of repetition.